appcues/src/main/java/com/appcues/RenderContextManager.kt

```diff
@@ -0,0 +1,26 @@
+package com.appcues
+
+import android.view.ViewGroup
+import com.appcues.data.model.RenderContext
+import com.appcues.data.model.RenderContext.Embed
+import java.lang.ref.WeakReference
+
+internal class RenderContextManager {
+
+    private val slots: HashMap<RenderContext, WeakReference<ViewGroup>> = hashMapOf()
+
+    fun registerEmbedFrame(frameId: String, view: ViewGroup) {
+        slots[Embed(frameId)] = WeakReference(view)
+    }
+
+    fun getEmbedView(renderContext: RenderContext): ViewGroup? {
+        return when (renderContext) {
+            is Embed -> slots[renderContext]?.get().also { if (it == null) slots.remove(renderContext) }
```

Comment:
this is the part where we would not actually call `slots.remove()` yet, based on convo about cell re-use (recyclers), correct?

Reply:
good catch